src/lib/components/core/header/Header.svelte

```diff
@@ -145,14 +148,13 @@ async function openProject() {
 		mode.set(Mode.PYTHON);
 		robot.set(robots.l_nano_rp2040);
 		code.set(await content.text());
-	} else {
-		if (get(mode) === Mode.BLOCKS) {
-			loadWorkspaceFromString(await content.text(), $workspace);
-		} else {
-			restore.set(JSON.parse(await content.text()));
-			mode.set(Mode.BLOCKS);
-		}
-		robot.set(robots[file.name.split(".").at(-1)]);
+	} else if (file.name.endsWith(".py.json")) {
+		mode.set(Mode.PYTHONBLOCKS);
+		robot.set(robots.l_nano_rp2040);
+		loadWorkspaceFromString(await content.text(), $workspace);
+	} else if (file.name.endsWith(".json")) {
```

Comment:
no, the robot type is insanely important, it should be pulled from the file extension, this will break all existing projects and cause the differences in available blocks between for example flitz and original to cease to exist